### DISCUSSION TITLE
using std::move with emplace/push_back

### DISCUSSION THREAD

**Discussion by tawmoto** [nlohmann/json#2804] [Q&A] [answered]
Hello,
I have this json containing an array

```
Json json{};
Json jsonAnimal = {{"name", "dog"}, {"age", 20}};

json["animals"].emplace_back(std::move(jsonAnimal));
```
Is it ok to use `std::move` on the last line, like I would do in a normal STL container?

Thanks

**Comment by nlohmann** [comment, marked as answer]
Yes, this is fine. Moved-from values will be `null`.

**Reply by tawmoto** [reply]
Great thanks!

**Reply by tawmoto** [reply]
Also I guess, this can be done also, not sure if it's more efficient
`json["animals"].emplace_back(Json{{"name", "dog"}, {"age", 20}});`

It seems that this does not compile:
`json["animals"].emplace_back({{"name", "dog"}, {"age", 20}});`
